src/main/ngapp/accio/src/app/mask.directive.ts

```diff
@@ -0,0 +1,142 @@
+import { Directive, HostListener, ElementRef, Input } from '@angular/core';
+import { MagicWandService } from './magic-wand.service';
+
+@Directive({
+  selector: '[appMask]',
+})
+export class MaskDirective {
+  @Input() scale: number;
+  @Input() originalData: ImageData;
+  @Input() scaledData: ImageData;
+  @Input() maskData: ImageData;
+
+  // create direct reference of canvas on editor.html
+  constructor(
+    private canvas: ElementRef<HTMLCanvasElement>, 
+    private magicWandService: MagicWandService,
+  ) { }
+  private ctx: CanvasRenderingContext2D;
+  private maskImage: HTMLImageElement;
+
+  @HostListener('click', ['$event'])
+  onClick(e: MouseEvent) {
+    const xCoord = e.offsetX;
```

Comment:
Since typescript supports types you'll want to use them.

`const xCoord: number = e.offsetX;`

Here and below.

Reply:
I thought in the style guide in TS, when you initialize the variables when they are declared you didn't need to define the types?

Reply:
Oh you're right. I just looked at the style guide.